app/src/main/java/com/example/notes/model/NoteAdapter.kt

```diff
@@ -0,0 +1,55 @@
+package com.example.notes.model
+
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import android.widget.Toast
+import androidx.recyclerview.widget.RecyclerView
+import com.example.notes.R
+
+class NoteAdapter(
+    private val onNoteClick: (NoteData) -> Unit,
+    private val onNoteLongClick: (NoteData) -> Unit
```

Comment:
Instead of passing single callbacks, can use interface and define there as much as you want.
easier to maintain in the future.

Reply:
whats the meaning by that?
do you mean an interface of callbacks, which would be defined for the creation of the adapter, that way it would be more flexible for a different number of callbacks and such?